### DISCUSSION TITLE
[Help]: Is there a way to send an z-wave Alarm Cancellation Message ?

### DISCUSSION THREAD

**Discussion by dionmes** [zwave-js/node-red-contrib-zwave-js#137] [Q&A]
### How can we help?

First off, Awesome project. I first used the OpenZWave implementation but this is so much better.
So I have not used my zwave nodes like lights, dimmers etc. as alarm nodes. This because I do not know how to cancel a wave alarm. I know it is possible for zwave nodes to send and/or receive an alarm cancellation, but if I want to create my own manual switch for this (for example via the node-red Dashboard UI), I do not know what type of message to sent.
Anyone got any ideas on this ?

### Any more details?

A node-red Home Automation implementation using z-wave nodes, Phillips Hue, wifi devices and media equipment. Dashboard UI and Homekit used to control.

### Version

6.1.1

### Node-RED Version

2.0.6

### What hardware are you using?

Raspberry Pi

### Any code to add?

```shell
NA
```

**Comment by marcus-j-davies** [comment]
HI @dionmes ,

Are you referring to a z-wave siren of sorts?
we first need to define what your 'alarm' is, is it a z-wave siren of some kind?

or are you using your z-wave lights as a strobe of some kind?
I use a z-wave siren, and controlling it is a breaze.

I would also update to **6.2.0** - this has some killer features that allows to streamline all your commands with incredible ease!

**Comment by dionmes** [comment]
Hi Marcus, going to update this weekend. So the Alarm I am talking about is the specific Alarm Command Class (0x71). A node can generate such an alarm, such as a door sensor or motion sensor, and if I understand correctly this will be broadcasted. And nodes configured to receive this command class like lights and other will respond (like lights connected to a Fibaro zwave switch which will start blinking) . You can see this in the z-wave device configuration. 
So I am able to trigger the Alarm via a sensor, and devices will respond, but I do not know how to send an Alarm Cancellation Messages (so these lights will stop blinking, Siren will stop sounding etc.).

**Comment by marcus-j-davies** [comment]
So,

The CC **0x71** and the Reference  to **Alarm** was deprecated by Zwave Alliance quite some time ago.
**0x71** is now called **Notification** - and its used for many types of notifications.

There are 1000 ways to achieve what you want:  Easy -> Not So Easy

NOTE: I am creating the commands manually here, but the new CMD-Factory (6.2.0) - will construct them for you.

**Triggering:**

1) Just listen for your door being opened or motion being triggered within Node-Red
     - When that happens - send a command to your lights (presumably they have a flash function?)
     - If your lights depend on a notification, send the expected `event`  and `type` they want, in order to trigger said strobe.
     
  ```javascript
    {
         "mode": "CCAPI",
          "cc": "Notification",
          "method": "sendReport",
          "params": [
               {
                  "notificationType": 7, /* Home Security */
                  "notificationEvent": 1 /* Intrusion */
               }
           ]
       }
```

2) Use Associations
Whilst this works, both devices needs to support each others output messages
you will use the UI to set this up - it can be very simple or extremely annoying - really depends on your devices

**Stopping:**

**If Using 1:**
 just send the commands to stop everything (usually the Binary CC) - but it depends on the devices
and if your lights need a notification to stop

  ```javascript
    {
         "mode": "CCAPI",
          "cc": "Notification",
          "method": "sendReport",
          "params": [
               {
                  "notificationType": 7, /* Home Security */
                  "notificationEvent": 0 /* Idle */
               }
           ]
       }
```

I am using Home Security - but there are loads

**If Using 2:**
I would have thought the associations would take care of this


**How I do it (I only have a siren):**

When my motion is triggered I send this to my siren
  ```javascript
    {
         "mode": "CCAPI",
          "cc": "Notification",
          "method": "sendReport",
          "params": [
               {
                  "notificationType": 7, /* Home Security */
                  "notificationEvent": 1 /* Intrusion */
               }
           ]
       }
```

when I want to stop the alarm, I send this (but I could use idle in a notification - but have never really tried it with my siren).
```javascript
    {
         "mode": "CCAPI",
          "cc": "Binary Switch",
          "method": "set",
          "params": [false]
       }
```

see attached notifications types and events

[SDS13713-Notification-Command-Class.xlsx](https://github.com/zwave-js/node-red-contrib-zwave-js/files/7352953/SDS13713-Notification-Command-Class.xlsx)


What notification types/events you devices support will be in their manual, as well as there supported CC's obviously.

**Reply by marcus-j-davies** [reply]
@dionmes ,
I have moved this to a discussion (hope you don't mind)

**Reply by dionmes** [reply]
Great info Marcus, thx. 
So to me it seemed as if the notification message for an alarm is a broadcast instead of an association. From that perspective I do not have to sent an individual message to each node, but just a cancel notification broadcast. And that was what I was looking for, but I can be mistaken here.
I had the idea that is a sensor sends an alarm notification and all the nodes in the network respond to it (even the not directly associated in a group). But I do not want to try this to often because cancelling is such a burden. 
But now I might try it by creating that cancel message and wire it to all the nodes.

**Reply by marcus-j-davies** [reply]
You're almost there.

Here are the possibilities:

**Associations**: One device sends to other(s)  (software/You are not involved - BUT it is to setup the associations prior)
This is quite advanced - not always the easiest (but is supported in the UI)

**Multicast**: The software/you simply send **ONE**  message to the nodes you want to stop
Can be quite fiddly in some cases - I.e the command you are sending must be supported by target devices

**Simple Message per Target Node(s)**
Simply send a command for each target node type (lights, sirens) - a device node per group

The latter is by-far the easiest / less headache way
Its best to picture it (My example assumes the same commands are supported by my lights and Siren)

and its using 6.2.0 - with CMD-Factory

![image](https://user-images.githubusercontent.com/55892693/137504619-4ccb8bc6-61e8-46c0-a8b8-ac4a80186d63.png)

**Reply by dionmes** [reply]
Thank you again for time you put in my question. Maybe the per-device approach might be the best one. But the :

_Multicast: The software/you simply send ONE message to the nodes you want to stop
Can be quite fiddly in some cases - I.e the command you are sending must be supported by target devices_

Is actually the essence of my question, how do you create such a message and where would you sent it to ? But I will probably go to the per device approach.

**Reply by marcus-j-davies** [reply]
There is no special message - you set the device node to `Multicast` - and choose the devices you want to target.
the message will then be sent to those devices using Multicast.

But devices need to all support the passed in command - else it wont work

**Comment by crxporter** [comment]
I just want to come in real quick and say:

***Use the command factory node***

All those complicated JSON messages that marcus loves so much... will be created for you!!

**Reply by marcus-j-davies** [reply]
2nd it 😉

**Reply by dionmes** [reply]
Now I am curious, gonna look at it this weekend.
